libc/src/string/memory_utils/op_x86.h

```diff
@@ -25,6 +25,11 @@
 #include <immintrin.h>
 #endif
 
+// Disable GCC complaining about missing attributes when using SIMD types in
+// template specializations.
```

Comment:
Which attributes is it complaining about? Mind adding that to the commit message?

Reply:
It doesn't say which ones ☹
```
[17/29] Building CXX object projects/libc/test/src/__support/FPUtil/CMakeFiles/libc.test.src.__support.FPUtil.fpbits_test.__unit__.__build__.dir/fpbits_test.cpp.o
In file included from /redacted/gchatelet/git/llvm-project/libc/src/string/memory_utils/x86_64/inline_memcpy.h:15,
                 from /redacted/gchatelet/git/llvm-project/libc/src/string/memory_utils/inline_memcpy.h:22,
                 from /redacted/gchatelet/git/llvm-project/libc/src/__support/CPP/string.h:14,
                 from /redacted/gchatelet/git/llvm-project/libc/src/__support/FPUtil/fpbits_str.h:12,
                 from /redacted/gchatelet/git/llvm-project/libc/test/src/__support/FPUtil/fpbits_test.cpp:10:
/redacted/gchatelet/git/llvm-project/libc/src/string/memory_utils/op_x86.h:122:37: warning: ignoring attributes on template argument ‘__m128i’ [-Wignored-attributes]
  122 | template <> struct is_vector<__m128i> : cpp::true_type {};
      |                                     ^
/redacted/gchatelet/git/llvm-project/libc/src/string/memory_utils/op_x86.h:123:44: warning: ignoring attributes on template argument ‘__m128i’ [-Wignored-attributes]
  123 | template <> struct cmp_is_expensive<__m128i> : cpp::true_type {};
      |                                            ^
/redacted/gchatelet/git/llvm-project/libc/src/string/memory_utils/op_x86.h:162:37: warning: ignoring attributes on template argument ‘__m256i’ [-Wignored-attributes]
  162 | template <> struct is_vector<__m256i> : cpp::true_type {};
      |                                     ^
/redacted/gchatelet/git/llvm-project/libc/src/string/memory_utils/op_x86.h:163:44: warning: ignoring attributes on template argument ‘__m256i’ [-Wignored-attributes]
  163 | template <> struct cmp_is_expensive<__m256i> : cpp::true_type {};
      |                                            ^

```

Reply:
Looking at `__m128i` definition
```
/usr/lib/llvm-16/lib/clang/16/include/emmintrin.h:typedef long long __m128i __attribute__((__vector_size__(16), __aligned__(16)));
```

Reply:
I tried a better explanation, let me know what you think.

Reply:
ah, yeah https://stackoverflow.com/questions/41676311/implication-of-gcc-warning-ignoring-attributes-on-template-argument-wignored has more context.  I was just looking at /usr/lib/gcc/x86_64-linux-gnu/13/include/avxintrin.h myself.

Do you mind mimimizing the region of code for which the pragma is disabled? I think the initial push+disable could be "sunk" lower in the header.  That might help avoid false negatives that result from disabling the diagnostics.

Reply:
Sure thing. Done!